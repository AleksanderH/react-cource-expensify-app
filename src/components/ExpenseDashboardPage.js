import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseLIstFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;