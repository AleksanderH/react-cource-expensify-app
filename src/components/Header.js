import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux'
import { startLogout } from '../actions/auth';
 
export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <p><NavLink to='/' activeClassName='is-active' exact={true}>Dashboard</NavLink></p>
        <p><NavLink to='/create' activeClassName='is-active'>Create</NavLink></p>
        <p><NavLink to='/edit' activeClassName='is-active'>edit page</NavLink></p>
        <p><NavLink to='/help' activeClassName='is-active'>help</NavLink></p>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);