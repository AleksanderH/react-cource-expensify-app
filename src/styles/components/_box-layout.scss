.box-layout {
    background-image: url("/images/bg.jpg");
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    width: 100vw;
}

.box-layout__box{
    background: fade-out($color: #fff, $amount: .30);
    border-radius: 5px;
    padding: $xl-size $l-size;
    text-align: center;
    width: 35rem;
}

.box-layout__title{
    margin : 0 0  $m-size 0;
    line-height: 1;
}
