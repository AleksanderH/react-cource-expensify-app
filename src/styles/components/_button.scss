.button{
    margin: 0 0;
    background-color: $button-bg;
    color: $light;
    border: none;
    border-radius: 3px;
    padding: $m-size;
    font-size: $font-size-large;
    font-weight: 300;
    display: inline-block;
    text-decoration: none;

    transition: background-color 0.3s;
}

.button--link{
    background: transparent;
}
