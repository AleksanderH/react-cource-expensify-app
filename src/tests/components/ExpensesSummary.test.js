import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpenseSummary';

test('shoud correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={123} />);
    expect(wrapper).toMatchSnapshot();
});

test('shoud correctly render ExpensesSummary with multiple eexpenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={3446756756} />);
    expect(wrapper).toMatchSnapshot();
});