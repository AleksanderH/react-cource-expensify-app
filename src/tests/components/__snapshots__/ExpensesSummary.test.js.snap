// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud correctly render ExpensesSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaling 
      <span>
        $1.23
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button "
        replace={false}
        to="/create"
      >
        Add cost
      </Link>
    </div>
  </div>
</div>
`;

exports[`shoud correctly render ExpensesSummary with multiple eexpenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totaling 
      <span>
        $34467567.56
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button "
        replace={false}
        to="/create"
      >
        Add cost
      </Link>
    </div>
  </div>
</div>
`;
